### PR TITLE
Improve golang webapp interactive quickstart

### PR DESCRIPTION
**Pull request by sergiught**
Following some [captured feedback from customers](https://auth0.slack.com/archives/C3GQG00MS/p1658476001753869), we're aiming at clarifying some more where each file needs to be saved within this PR and also moving the "Serve your app" section last and adding how to run everything.

### CODE REVIEW

**Review by Widcket — APPROVED** [review]

**Comment by Widcket** [comment]
Is there a way to have the code files in the QS work as a kind of minimal sample app on their own? To avoid much of the folder structure, and have the files shown represent a valid app by themselves.

**Comment by sergiught** [comment]
@Widcket We could decide to flatten all the go files and not use any packages as well. However the reason for the package structure was to have more idiomatic code and good app design. 

There are also links that point directly to the github repo for the sample, so I found it odd that the structure wasn't clear.

**Comment by Widcket** [comment]
Just keep in mind that the code in the QS !== QS sample app. Of course, this approach with packages is completely valid.

**Comment by Widcket** [comment]
E.g. https://auth0.com/docs/quickstart/native/flutter and https://auth0.com/docs/quickstart/native/ios-swift/interactive feature snippets that form kind of a reduced, minimal sample app, which is different from the actual sample app

**Comment by sergiught** [comment]
That's good to know thanks @Widcket, I was under the assumption that the 2 should match as much as possible.

I would still however merge this in as it's an n+1 improvement and then consider simplifying further the QS in case we keep getting feedback that it's hard to follow, wdyt?

**Comment by Widcket** [comment]
That's a good idea.

**Review by frederikprijck — APPROVED** [review]
